src/samplebuffer.h

```diff
@@ -0,0 +1,77 @@
+#ifndef SAMPLEBUFFER_H
+#define SAMPLEBUFFER_H
+
+#include "util/types.h"
+
+#include <algorithm> // std::swap
+
+// A sample buffer with properly aligned memory to enable SSE optimizations.
+// The public SINTerface closely follows that of std::vector.
```

Comment:
mass renaming: SINTerface
"closely follows that of std::vector": does this add valuable info?  Outdated?